### PR TITLE
More print only rank0 and removing virtual get_comm()

### PR DESCRIPTION
**Pull request by tcclevenger**
More fixes for only output on rank 0. Requires ekat to update.

### CODE REVIEW

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp [hunk 1]
```diff
@@ -287,7 +287,7 @@ void HommeDynamics::initialize_impl (const util::TimeStamp& /* t0 */)
   const auto& Q_dyn_track = get_group_out("Q",dgn).m_bundle->get_header().get_tracking();
   for (auto wp : Q_dyn_track.get_providers()) { 
     auto sp = wp.lock();
-    printf("Q is provided by: %s\n",sp->name().c_str());
+    if (get_comm().am_i_root()) printf("Q is provided by: %s\n",sp->name().c_str());
```
Comment (bartgol) [inline components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:290]: I think I added this for loop while I was debugging (trying to figure out who else was computing Q). Can you please remove it altogether?

**Review by bartgol — CHANGES_REQUESTED** [review]
Looks good. I have a cleanup favor to ask though.

File: components/scream/src/mct_coupling/scream_cxx_f90_interface.cpp [hunk 2]
```diff
@@ -64,6 +64,11 @@ const scream::control::AtmosphereDriver& get_ad () {
   return c.get<scream::control::AtmosphereDriver>();
 }
 
+const ekat::Comm& get_ekat_comm () {
```
Comment (bartgol) [inline components/scream/src/mct_coupling/scream_cxx_f90_interface.cpp:67]: I think we should instead add `get_comm()` to the AtmosphereDriver class.

Comment (tcclevenger) [reply components/scream/src/mct_coupling/scream_cxx_f90_interface.cpp:67]: Yeah, I think there is a comment somewhere about removing every instance of `m_[insert-process]_comm` and instead opt for a base class comm. Then we would have a `AtmosphereDriver::get_comm()`. I can do that as a part of this PR.

**Comment by tcclevenger** [comment]
@bartgol I removed all the `m_[process name]_comm` and `m_[process name]_param` from `AtmosphereProcess` derived classes. Instead now there are 2 `AtmosphereProcess` constructors: default constructor and one that takes `comm, params` as input. Each derived class has the option to call 
```
DerivedClass (const ekat::Comm& comm, const ekat::ParameterList& params)
  : AtmosphereProcess(comm, params)
{ ... }
```
Now only `AtmosphereProcess` has `get_comm()`.

File: components/scream/src/share/atm_process/atmosphere_process.hpp [hunk 3]
```diff
@@ -72,6 +72,12 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
   template<typename T>
   using str_map = std::map<std::string,T>;
 
+  // Default base constructor
+  AtmosphereProcess () = default;
```
Comment (bartgol) [inline components/scream/src/share/atm_process/atmosphere_process.hpp:76]: Question: do we need/want a default ctor? Who uses it?

**Review by bartgol — COMMENTED** [review]
Looks good. Only one question regarding atm proc ctor.

File: components/scream/src/share/atm_process/atmosphere_process.hpp (hunk 3 above)
Comment (tcclevenger) [reply components/scream/src/share/atm_process/atmosphere_process.hpp:76]: No one currently. My only thought was if a process (or test) doesn't care about params and comms then they would only have to define their own constructor. But maybe this is not something that would happen. I can remove it.

Comment (bartgol) [reply components/scream/src/share/atm_process/atmosphere_process.hpp:76]: Uhm, I think devs of such an atm proc would not have a hard time creating dummy comm/params. OTOH, if someone accidentally uses the def ctor, there might be issues finding the bug. I think we can remove it, and if need be, add it back later.

**Comment by bartgol** [comment]
I believe the failure we see on weaver is already in master (I manually verified that master can fail on weaver1). I will merge this.

**Review by bartgol — APPROVED** [review]

**Comment by tcclevenger** [comment]
Thanks @bartgol!

**Pull request merged**